src/ext/java/org/opentripplanner/ext/siri/updater/azure/AbstractAzureSiriUpdater.java

```diff
@@ -186,29 +191,38 @@ public boolean isPrimed() {
     return this.isPrimed;
   }
 
-  public void setPrimed(boolean primed) {
-    isPrimed = primed;
+  private void setPrimed() {
+    isPrimed = true;
   }
 
   @Override
   public String getConfigRef() {
     return this.configRef;
   }
 
-  protected String fetchInitialData(URI uri) {
-    // Maybe put this in the config?
-    HttpHeaders rh = HttpHeaders.of().acceptApplicationXML().build();
-    String initialData;
+  /**
+   * Returns None for empty result
+   */
+  protected Optional<ServiceDelivery> fetchInitialSiriData(URI uri) {
+    var headers = HttpHeaders.of().acceptApplicationXML().build().asMap();
+
     try (OtpHttpClient otpHttpClient = new OtpHttpClient()) {
-      initialData =
-        otpHttpClient.getAndMap(
-          uri,
-          Duration.ofMillis(timeout),
-          rh.asMap(),
-          is -> CharStreams.toString(new InputStreamReader(is))
-        );
+      var t1 = System.currentTimeMillis();
+      var siriOptional = otpHttpClient.executeAndMapOptional(
+        new HttpGet(uri),
+        Duration.ofMillis(timeout),
+        headers,
+        SiriXml::parseXml
+      );
+      var t2 = System.currentTimeMillis();
+      LOG.info("Fetched initial data in {} ms", (t2 - t1));
+
+      if (siriOptional.isEmpty()) {
+        LOG.info("Got status 204 'No Content'.");
+      }
+
+      return siriOptional.flatMap(siri -> Optional.ofNullable(siri.getServiceDelivery()));
```

Comment:
```suggestion
      return siriOptional.map(siri -> siri.getServiceDelivery());
```

You don't actually need the flatMap. Optional#map is smart enough to convert a null into an empty Optional.

```
System.out.println(Optional.of("henrik").map(s -> null))
-> Optional.empty()
```